docs/coding-guidelines.md

```diff
@@ -353,10 +353,18 @@ or report a warning back to the collector with a clear error saying CGO is requi
 ## Breaking changes
 
 Whenever possible, we adhere to [semver](https://semver.org/) as our minimum standards. Even before v1, we strive not to break compatibility
-without a good reason. Hence, when a change is known to cause a breaking change, it MUST be clearly marked in the
-changelog and SHOULD include a line instructing users how to move forward.
+without a good reason. Hence, when a change is known to cause a breaking change, we intend to follow these principles:
 
-We also strive to perform breaking changes in two stages, deprecating it first (`vM.N`) and breaking it in a subsequent
+- Breaking changes MUST have migration guidelines that clearly explain how to adapt to them.  
+- Users SHOULD be able to migrate to the breaking change at their own pace, independent of other software updates.  
```

Comment:
Can we define "software updates"? Is this related to other updates within the Collector specifically, changes to its environment, or changes to software it communicates with?

Reply:
Replaced by "Collector updates"